### DISCUSSION TITLE
is adding pdf annotations usefull?

### DISCUSSION THREAD

**Discussion by kaiserbene** [jonaswinkler/paperless-ng#266] [Feature Requests]
when scanning my documents of the last 10 years i wondered how often i wrote notes on my docs. As importing docs sent by mail directly to paperless from now on, I'd like to put annotations on the docs - i.e. "paid at 2020/12/27". Or to put stamps on the document. Do you think that would be usefull? I have no idea how to realize something like that but I've seen some web based annotators by googeling that

**Comment by jonaswinkler** [comment]
Putting 'stamps' or annotations on documents directly (that is, edit the PDF directly) has some drawbacks: You can't search or filter for that information, it is unstructured, and it modifies the original documents (which I'm trying to avoid, don't want to mess up or potentially destroy documents due to some failed edits).

Someone had the idea about adding *custom metadata* to paperless in #274. Example: With this, you could:
- Tell paperless that you want to store a "Paid on" date for documents marked with the document type "invoice"
- This would add a new field to the document editor (below correspondent, tags, etc) when the document type is "invoice"
- Save "Paid on" for invoices.

If that sounds useful, please comment over there.

I'm not sure if that's useful. To me, it sounds like a lot of manual editing with little benefits. Most notably, this will most likely not help you find documents quicker.

**Reply by kaiserbene** [reply]
You're right in all points, for me an addtitional text-field for annotations would suite my needs. But having them categorized and structured would be fine, too. Maybe a bit of an overkill and slightly too complex to use...

**Reply by jonaswinkler** [reply]
Maybe an additional free form, multi line "notes" field is all we need.

**Reply by totti4ever** [reply]
thinking of it, this actually seems to meet pretty much what we have when annotating stuff on free space of a document before scanning: a multi-line note "field" :-)

**Reply by Burschi500** [reply]
I very much like the idea of having a multiline text-field, or even custom metadata, as i too noted the "paid xx.xx.xxxx" on many documents. As long as its not available one can come along with editing the "content" field, which is searchable. One can even add tags in that field to find them, like X-Custom-Note, but that needs some selfcontrol ;)

**Comment by kaiserbene** [comment]
👍

**Comment by Pamalosebi** [comment]
Personally I think that's a great idea. My usecase would be to sign papers with my signature digitally. That would save the download and sort it again local just to upload it again.

I understand the concern that the original should never be touched but honestly... Just make a copy or versioning. One could make a "modified" directory next to the archive and push such files in there (within the paperless structure of cause).

To have that feature would be fantastic.

**Comment by TheNomad11** [comment]
Could be very useful. One could use paperless-ng in academic settings thdn, to store academic articles with notes, quotes and thoughts, and use it for knowledge management (although this might not have been the original idea of this project, I suppose)

**Comment by e-patrick** [comment]
Thinking of what I do with my paper documents - and what I would need to be able to do with PDF files in a true paperless office - leads me to the opinion that editing the document itself is just inevitable. I do often use a highlighter to mark important parts, I scribble random thoughts or notes from a releated phone call on it, usually close to the parts I refer to, and so on. 
Yes, this is not searchable and not structured and it doesn't help me to find my documents more quickly - these are all valid points. But in the end, when I eventually have found my document, it provides a lot of useful information.
So I do fully support the idea of making the document editable, and also the idea of having some sort of versioning, keeping the original and creating a copy to be edited.

Having the edit capability would therefore make an already fantastic product even better :)

**Reply by ffe4** [reply]
I agree with @e-patrick that editing the document itself (a non-destructive way), would be an essential feature for many use cases.

For example when it comes to contracts, highlighting important parts, making notes for clarification, or marking parts as superseded by supplemental agreements are all things I would like to be able to do. Right now, for every change, be it a new comment or correcting a typo in your latest comment, you have to download the pdf, make your edits, re-upload it, and then decide whether you want to also store the old version(s), all while making sure you actually edited the version with the latest edits.

While annotations are probably not an important day-to-day feature for most people, they are particularly useful for long-term storage, where you might want to fill in your future self on some details.
